src/actions/advancement.cpp

```diff
@@ -54,6 +55,17 @@ static lg::log_domain log_display("display");
 
 namespace
 {
+	template<typename T>
+	std::vector<T> vector_intersection(const std::vector<T>& a, const std::vector<T>& b)
```

Comment:
Why not just use `std::set_intersection`? (For that you'll need to sort the vector first, but I doubt that would cause any issues.)

Reply:
hmm i i removed the code which used that function, so i shoudl probalby just remove that aswell.
but to answer your question: as you said, std::set_intersection required me to sort the vector, the whole point of the function is to make getting an intersection of  two vectors a one liner, in particular to not clutter the other code too much with how exactly i get the intersection of two vectors.